image/cas/interface.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cas implements generic content-addressable storage.
+package cas
+
+import (
+	"io"
+)
+
+// Engine represents a content-addressable storage engine.
+type Engine interface {
+
+	// Put adds a new blob to the store.
+	Put(writer io.Reader) (digest string, err error)
```

Comment:
Since there isn't mediaType information provided in put and returned by get, should the user autodetect the mediaType or is this just not needed since it'll be already known because the user will always fetch a blob starting from a parent referencing object (descriptor, manifest, manifest list) that already provides the mediaType?

This assumes that all the objects will be part of a tree and there's always the way to retrieve the mediaType by the parent.

Instead if there isn't this assumption then a mediaType should be returned or it should be detected by the user for every object. (another example can be a List method that can return blobs by mediaTypes)